# This file is part of HiGFXback

# requires
REQUIRES="libmad-build make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=

# build
make

# install
install -d $DESTDIR/bin
install minmad $DESTDIR/bin

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/minmad-build.pc << EOF
Name: minmad
Version: 20220205
Description: Minimalist libmad-based MP3 player
Requires: $REQUIRES

exec=\\
/bin/minmad
EOF

exit

# patch
--- minmad-20220205.orig/Configure
+++ minmad-20220205/Configure
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
--- minmad-20220205.orig/Makefile
+++ minmad-20220205/Makefile
@@ -1,3 +1,4 @@
+PREFIX = /usr/local
 CC = cc
 CFLAGS = -Wall -O2
 LDFLAGS = -lmad
@@ -9,4 +10,5 @@
 	$(CC) $(LDFLAGS) -o $@ $^
 clean:
 	rm -f *.o minmad
-
+install: all
+	install -D minmad $(DESTDIR)$(PREFIX)/minmad

# source
https://github.com/aligrudi/minmad/archive/07062b35ab0a25bbc97abe64327dca6a5021b750/minmad-20220205.tar.gz
